js/convert.js

```diff
@@ -0,0 +1,177 @@
+var fs = require('fs');
+var glob = require('glob');
+var linebyline = require('n-readlines');
+var cmd = require('node-cmd');
+var exec = require('child_process').exec;
+
+
+var path = '/Users/psuc44/dev/shaun/examples/';
+
+function listFiles() {
+    var filelist = [];
+    var files =  glob.sync(path + '**').filter(isFile);
+
+    for (var i = 0; i < files.length; i++) {
+        if (getFilename(files[i]).split('.')[1] == 'elm') {
+            if (hasView(files[i]) == true) { 
+                filelist.push(files[i]);
+            }
+        }
+    }   
+    return filelist;
+}
+
+function generatePort(module_name) {
+    var port_name = portName(module_name);
+    return `
+port ${port_name} : String 
+port ${port_name} =
+    ${module_name}.view 
+        |> render`
+}
+
+function generateImport(module_name) {
+    return `import ${module_name}`
+}
+
+function portName(module_name) {
+    return module_name.replace('.','').toLowerCase();
+}
+
+function fileName(module_name, basedir, with_lower) {
+
+    if (basedir == null) {
+        basedir = './';
+    } 
+    var with_basedir = basedir + module_name.replace('.','/');
+
+    if (with_lower == true) {
+        return with_basedir.toLowerCase();
+    } return with_basedir;
+}
+
+function generateMapping(port, file) {
+    return `echo "fs.writeFile('${file}.html', elm.ports['${port}']);" >> _main.js`
+}
+
+function generate_vdom(module_names, basedir) {
+
+    var port_files = []
+
+    for (var i = 0; i < module_names.length; i++) {
+        port_files.push({
+            'port' : portName(module_names[i]),
+            'filename' : fileName(module_names[i], basedir, false)
+        });
+    }
+
+    var ports = module_names.map(generatePort).join('\n');
+    var imports = module_names.map(generateImport).join('\n');
+    var maps = [];
+
+    var port_file_values = []
+    
+    for (var i = port_files.length-1; i >= 0; i--) {
+
+        var curr = port_files[i];
+
+        maps.push(generateMapping(curr.port, curr.filename));
+
+        port_file_values.push(curr.filename)
+
+    }
```

Comment:
this block could be `var port_file_values = port_files.map(function(curr){...})` instead